### PR TITLE
chore(eslint-plugin): stop linting rule output formatting in tests

### LINKED ISSUE #4911: @typescript-eslint/internal/plugin-test-formatting too strict about snapshot indentation
Complained in https://github.com/typescript-eslint/typescript-eslint/pull/4901#discussion_r864456822

Take this code for example (from that PR):

```ts
{
      code: `
class Data {
  member<T extends unknown>() {}
}
      `,
      errors: [
        {
          data: { constraint: 'unknown', name: 'T' },
          messageId: 'unnecessaryConstraint',
          endColumn: 27,
          column: 10,
          line: 3,
          suggestions: [
            {
              messageId: 'removeUnnecessaryConstraint',
              output: `
class Data {
  member<T>() {}
}
              `,
            },
          ],
        },
      ],
    },
```

The test fails, suggesting that:

```
Expected value to strictly be equal to:
      "
    class Data {
      member<T>() {}
    }
                  "
    Received:
      "
    class Data {
      member<T>() {}
    }
          "
```

Okay, let's try another way by indenting everything—maybe it will be dedented? (Spoiler: I don't know how the snapshot works.)

```ts
{
      code: `
        class Data {
          member<T extends unknown>() {}
        }
      `,
      errors: [
        {
          data: { constraint: 'unknown', name: 'T' },
          messageId: 'unnecessaryConstraint',
          endColumn: 35,
          column: 18,
          line: 3,
          suggestions: [
            {
              messageId: 'removeUnnecessaryConstraint',
              output: `
                class Data {
                  member<T>() {}
                }
              `,
            },
          ],
        },
      ],
    },
```

```
    - Expected
    + Received


    -                 class Data {
    -                   member<T>() {}
    -                 }
    -               
    +         class Data {
    +           member<T>() {}
    +         }
    +       
```

o_o

It turns out the only way to work around it is by either fighting the linter and adding ignore comments, or by using `.trimEnd()` to each snapshot.

I think it's fine for the snapshots to be formatted nicely, so what about a sanitation step that removes all extraneous indentation, or removes trailing spaces (like `.trimEnd()`)? I think ESLint's tests does that.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

-   [x] Addresses an existing open issue: fixes #4911
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Inside the `@typescript-eslint/internal/plugin-test-formatting` rule, no longer checks formatting for `output` properties. 

Removes `noFormat`, `.trim()`, `.trimEnd()`, and `.trimRight()` from test outputs.

### CODE REVIEW

File: packages/eslint-plugin/wat.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+yarn run v1.22.19
+$ /Users/josh/repos/typescript-eslint/node_modules/.bin/jest --json
```
Comment (bradzacher) [inline packages/eslint-plugin/wat.txt:2]: should this be delet???

**Review by bradzacher — APPROVED** [review]

File: packages/eslint-plugin/wat.txt (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/wat.txt:2]: wat

Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/wat.txt]: I swear, the number of times this has happened, I should just check in `wat*` as a `.gitignore` rule to all the repos I work on.

**Review by bradzacher — APPROVED** [review]

**Comment by bradzacher** [comment]
image

**Pull request merged**